funcs/func_groupcount.c

```diff
@@ -1442,29 +1443,23 @@ static int group_channel_list_function_read(struct ast_channel *chan, const char
 				break;
```

Comment:
I think there should be a warning emitted if all the channels don't fit into the buffer, rather than silently truncating.
Other than that, looks good!

Reply:
Thanks!

This is why I was thinking about having support for a structured data return like a hash.  We should be able to put an 'unlimited' amount of things into a hash in theory.

But of course, HASHKEYS is going to have the same limit that any other dialplan return is going to have, which is the fixed-size of buf

Reply:
I don't actually think it's likely to be an issue... nobody is going to have 1,000 channels in a group... however, just to cover all bases, there should be a warning in case something weird happens, is all I care about.

Reply:
This is why I previously implemented these functions as iterator-style where you have a START/NEXT/END and then you can keep asking for the next one until you run out.  Thereby avoiding any kind of buf-size limit... but you didn't like that style of functions because nothing else in asterisk-dialplan behaves like this.  With > 400 channels (or just really long channel names like Local/ channels can be)... you can easily run into hitting the upper limit on the size of buf

Reply:
Right, again, I'm saying that I don't think it's a real issue - all I want is a warning in the case that should never be hit anyways :)

Reply:
I think adding HASHKEYS_ITERATOR and adding an option to all the things that return a list to be able to also return a HASH would solve this issue

Reply:
Look, my issue is not that this might be truncated. I don't have any use case for retrieving so many channels the buffer would be exhausted. I just think there needs to be a warning there, that's all.

Please don't over-complicate this... just add a warning so I can approve this PR

Reply:
Yup, I'll add in a warning and play with adding a HASHKEYS_ITERATOR.. it's something i've wanted anyway (separate project)

Reply:
```
    -- Executing [3333@services:1] Set("Local/3333@services-000002bb;2", "GROUP()=foo") in new stack
    -- Executing [3333@services:2] NoOp("Local/3333@services-000002bb;2", "groups=GROUP_CHANNEL_LIST(foo)") in new stack
[2024-03-18 11:21:17.074-0400] WARNING[943161][C-000004b0]: func_groupcount.c:1445 group_channel_list_function_read: GROUP_CHANNEL_LIST(foo) Channel membership list too large for buffer, truncating.
    -- Executing [3333@services:3] Set("Local/3333@services-000002bb;2", "groups=Local/3333@services-000001f4;2,Local/3333@services-000001f5;2,Local/3333@services-000001f6;2,Local/3333@services-000001f7;2,Local/3333@services-000001f8;2,Local/3333@services-000001f9;2,Local/3333@services-000001fa;2,Local/3333@services-000001fb;2,Local/3333@services-000001fc;2,Local/3333@services-000001fd;2,Local/3333@services-000001fe;2,Local/3333@services-000001ff;2,Local/3333@services-00000200;2,Local/3333@services-00000201;2,Local/3333@services-00000202;2,Local/3333@services-00000203;2,Local/3333@services-00000204;2,Local/3333@services-00000205;2,Local/3333@services-00000206;2,Local/3333@services-00000207;2,Local/3333@services-00000208;2,Local/3333@services-00000209;2,Local/3333@services-0000020a;2,Local/3333@services-0000020b;2,Local/3333@services-0000020c;2,Local/3333@services-0000020d;2,Local/3333@services-0000020e;2,Local/3333@services-0000020f;2,Local/3333@services-00000210;2,Local/3333@services-00000211;2,Local/3333@services-00000212;2,Local/3333@services-00000213;2,Local/3333@services-00000214;2,Local/3333@services-00000215;2,Local/3333@services-00000216;2,Local/3333@services-00000217;2,Local/3333@services-00000218;2,Local/3333@services-00000219;2,Local/3333@services-0000021a;2,Local/3333@services-0000021b;2,Local/3333@services-0000021c;2,Local/3333@services-0000021d;2,Local/3333@services-0000021e;2,Local/3333@services-0000021f;2,Local/3333@services-00000220;2,Local/3333@services-00000221;2,Local/3333@services-00000222;2,Local/3333@services-00000223;2,Local/3333@services-00000224;2,Local/3333@services-00000225;2,Local/3333@services-00000226;2,Local/3333@services-00000227;2,Local/3333@services-00000228;2,Local/3333@services-00000229;2,Local/3333@services-0000022a;2,Local/3333@services-0000022b;2,Local/3333@services-0000022c;2,Local/3333@services-0000022d;2,Local/3333@services-0000022e;2,Local/3333@services-0000022f;2,Local/3333@services-00000230;2,Local/3333@services-00000231;2,Local/3333@services-00000232;2,Local/3333@services-00000233;2,Local/3333@services-00000234;2,Local/3333@services-00000235;2,Local/3333@services-00000236;2,Local/3333@services-00000237;2,Local/3333@services-00000238;2,Local/3333@services-00000239;2,Local/3333@services-0000023a;2,Local/3333@services-0000023b;2,Local/3333@services-0000023c;2,Local/3333@services-0000023d;2,Local/3333@services-0000023e;2,Local/3333@services-0000023f;2,Local/3333@services-00000240;2,Local/3333@services-00000241;2,Local/3333@services-00000242;2,Local/3333@services-00000243;2,Local/3333@services-00000244;2,Local/3333@services-00000245;2,Local/3333@services-00000246;2,Local/3333@services-00000247;2,Local/3333@services-00000248;2,Local/3333@services-00000249;2,Local/3333@services-0000024a;2,Local/3333@services-0000024b;2,Local/3333@services-0000024c;2,Local/3333@services-0000024d;2,Local/3333@services-0000024e;2,Local/3333@services-0000024f;2,Local/3333@services-00000250;2,Local/3333@services-00000251;2,Local/3333@services-00000252;2,Local/3333@services-00000253;2,Local/3333@services-00000254;2,Local/3333@services-00000255;2,Local/3333@services-00000256;2,Local/3333@services-00000257;2,Local/3333@services-00000258;2,Local/3333@services-00000259;2,Local/3333@services-0000025a;2,Local/3333@services-0000025b;2,Local/3333@services-0000025c;2,Local/3333@services-0000025d;2,Local/3333@services-0000025e;2,Local/3333@services-0000025f;2,Local/3333@services-00000260;2,Local/3333@services-00000261;2,Local/3333@services-00000262;2,Local/3333@services-00000263;2,Local/3333@services-00000264;2,Local/3333@services-00000265;2,Local/3333@services-00000266;2,Local/3333@services-00000267;2,Local/3333@services-00000268;2,Local/3333@services-00000269;2,Local/3333@services-0000026a;2,Local/3333@services-0000026b;2,Local/3333@services-0000026c;2,Local/3333@services-0000026d;2,Local/3333@services-0000026e;2,Local/3333@services-0000026f;2,Local/3333@services-00000270;2,Local/3333@services-00000271;2,Local/3333@services-00000272;2,Local/3333@services-00000273;2,Local/3333@services-00000274;2,Local/3333@services-00000275;2,Local/3333@services-00000276;2,Local/3333@services-00000277;2") in new stack
    -- Executing [3333@services:4] NoOp("Local/3333@services-000002bb;2", "GROUP_CHANNEL_LIST_ERROR=TRUNCATED") in new stack

```